conda_smithy/templates/github-actions.yml.tmpl

```diff
@@ -38,66 +78,34 @@ jobs:
             SHORT_CONFIG: {{ data.short_config_name }}
         {%- endif %}
             UPLOAD_PACKAGES: {{ data.upload }}
-        {%- if data.build_platform.startswith("osx-64") %}
-            os: macos
-        {%- if github_actions.self_hosted %}
-            labels: ['macOS', 'self-hosted', 'x64']
-        {%- endif %}
-        {%- elif data.build_platform.startswith("osx-arm64") %}
-            os: macos
+            os: {{ platform_vars[data.build_platform]['os'] }}
+        {#- namespace needed to handle jinja scoping rules #}
+        {%- set ns = namespace(gpu=False) %}
         {%- if github_actions.self_hosted %}
-            labels: ['macOS', 'self-hosted', 'ARM64']
+            labels:
+              - self-hosted
+              - {{ platform_vars[data.build_platform]['self_hosted_label_os'] }}
+              {%- if platform_vars[data.build_platform]['self_hosted_label_arch'] %}
+              - {{ platform_vars[data.build_platform]['self_hosted_label_arch'] }}
+              {%- endif %}
+              {%- for label in github_actions.self_hosted_labels %}
+                {%- if 'gpu' in label %}
+                  {%- set ns.gpu = True %}
+                {%- endif %}
+                {%- if label.startswith('cirun-') %}
+                  {%- set label = label ~ "--${{ github.run_id }}-" ~ data.short_config_name %}
```

Comment:
We need unique labels so the on-demand creation mechanism works without issues. See https://docs.cirun.io/reference/unique-runner-labels.